### ISSUE TITLE
PTRACE_INTERRUPT raced with another event

### ISSUE THREAD

**Issue by BenjaminSchubert** [mozilla/rr#1480]
I've got a problem while running RR under docker on an old buggy version of transmission (v1.42). Here is what rr -V record gives me : 

[WARN wait() errno: 0 'Success'] PTRACE_INTERRUPT raced with another even 0x857f
[Warn done_preparing() errno: 0 'Success'] `epoll_wait' neede a scratch bugger of size 6291448, but only 2097152 was available. Disabling context switching : deadlock may follow.

This message keeps getting printed again and again. The exact same command did work on an older version of RR. (don't remember which though)

Do you have any idea of what it could be ?

Thanks

**Comment by rocallahan** [comment]
You mean *both* messages are printed again and again?

It's a little hard to believe that epoll_wait really needs a buffer that big. That's hundreds of thousands of events. Are the epoll_wait calls really doing that?

Are you recording 32-bit or 64-bit programs?

**Comment by BenjaminSchubert** [comment]
Yes I mean both, sorry

I cannot tell sadly :/ I know that running this in VmWare works, on the computer directly too, but not with docker..

It is a 64-bit programs.

**Comment by rocallahan** [comment]
Can you run the rr tests under docker? Do *any* rr tests run OK?

**Comment by BenjaminSchubert** [comment]
Yeah sure I will run the rr tests under docker ! And all my other programs do work well under docker and rr (apache, memcached, and a few more)

**Comment by BenjaminSchubert** [comment]
Ok, I ran the tests and most of them are failing. Here is the [log](https://www.dropbox.com/s/xeioxc7buwet7pv/rr.log?dl=0) . Any idea of what could make all of these fail ? I could try to fix some if I get a clearer direction, but for now I don't really understand

**Comment by rocallahan** [comment]
Sorry about the delay, I've been traveling. I'll look into this ASAP.

**Comment by rocallahan** [comment]
Do 32-bit programs run under Docker at all? E.g. does running `rr/obj/bin/simple_32` under Docker actually run and print `EXIT-SUCCESS` and terminate normally?

Does the `capget` test  (`rr/obj/bin/capget`) run under Docker, doing the same thing?
